leku/src/main/java/com/schibstedspain/leku/LocationPickerActivity.java

```diff
@@ -250,10 +252,15 @@ private void setUpFloatingButtons() {
     btnAcceptLocation.setOnClickListener(v -> returnCurrentPosition());
 
     FloatingActionButton btnSatellite = (FloatingActionButton) findViewById(R.id.btnSatellite);
-    btnSatellite.setOnClickListener(view -> {
-      map.setMapType(map.getMapType() == MAP_TYPE_SATELLITE ? MAP_TYPE_NORMAL : MAP_TYPE_SATELLITE);
-      btnSatellite.setImageResource(map.getMapType() == MAP_TYPE_SATELLITE ? R.drawable.ic_satellite_off : R.drawable.ic_satellite_on);
-    });
+    if(enableSataliteView) {
+      btnSatellite.setOnClickListener(view -> {
+        map.setMapType(map.getMapType() == MAP_TYPE_SATELLITE ? MAP_TYPE_NORMAL : MAP_TYPE_SATELLITE);
+        btnSatellite.setImageResource(map.getMapType() == MAP_TYPE_SATELLITE ? R.drawable.ic_satellite_off : R.drawable.ic_satellite_on);
+      });
+    }
+    else{
+      btnSatellite.setVisibility(View.GONE);
```

Comment:
Maybe `btnSatellite.setVisibility(enableSataliteView ? View.VISIBLE : View.GONE);` is enough